fix: Use correct CHAINID in EVM

DavePearce opened the pull request:
This updates the `ToyExecutionEnvironment` to use the correct CHAINID when constructing the EVM.  Previously, this was not done correctly and meant that the `CHAINID` bytecode was returning an invalid value, which in turn mean that storage addresses were being computed incorrectly (i.e. because they incorporated the CHAINID into the hash).

DavePearce commented:
@powerslider I am not so certain about changes to the `ToyExecutionEnvironmentV2`.  As far as I can tell, this doesn't read the `chainID` out of the `tx` like the original `ToyExecutionEnvironment`.

letypequividelespoubelles commented on zkevm-constraints:
why updating the constraint submodule ?

DavePearce replied:
Oh whoops, I will fix that (later today).

DavePearce replied:
Fixed now, and tests all passing.

DavePearce committed:
fix: Use correct CHAINID in EVM

This updates the `ToyExecutionEnvironment` to use the correct CHAINID
when constructing the EVM.  Previously, this was not done correctly and
meant that the `CHAINID` bytecode was returning an invalid value, which
in turn mean that storage addresses were being computed incorrectly
(i.e. because they incorporated the CHAINID into the hash).
